sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolvePartitionSpec.scala

```diff
@@ -50,6 +50,16 @@ object ResolvePartitionSpec extends Rule[LogicalPlan] {
           }
         case _ => command
       }
+    case command @ ShowTableExtended(ResolvedNamespace(catalog: TableCatalog, namespace),
```

Comment:
Could you clarify why `ShowTableExtended ` doesn't extend `V2PartitionCommand`. And apparently why do we need this special case.

Reply:
According to my understanding of the processing logic of V2PartitionCommand in ResolvePartitionSpec, 
the table it processes is a single table,

However, our ShowTableExtended supports two situations:
- (A).SHOW TABLE EXTENDED FROM/IN catalog.namespace LIKE 'table_name_pattern'; --- support a table regex pattern 
- (B).SHOW TABLE EXTENDED FROM/IN catalog.namespace LIKE 'table_name' PARTITION (part1 = 1); --- not support a table regex pattern, only support a certain table name.
https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-aux-show-table.html
<img width="738" alt="image" src="https://github.com/apache/spark/assets/15246973/95e62919-1777-4421-a759-c84626782e2b">

In case A, the tables it processes may be multiple tables, which is a regex pattern. It seems inappropriate to resolve the table in `ResolvePartitionSpec`.
In case B, the tables it processes is single (currently we only support the partitioned extended display information function in the case of one table)

If it is only a table name and partition information is specified, we can resolved partition here (covering situation B)
so made a special logic here.

Reply:
I think we need two commands for this SQL statement: `ShowTablesExtended` and `ShowTablePartition`. In `ShowTablePartition`, we can put `UnresolvedTable` in it and extend `V2PartitionCommand`

Reply:
Ok, I think it's also fine, Let me update it.

Reply:
Resolve ShowTableExtended here, looks weird.

Reply:
Based on the above suggestions, the modifications have been completed and we will add 2 commands: `ShowTablesExtended ` and `ShowTablePartition `
Simultaneously delete: `ShowTableExtended`
And in  `ResolveSessionCatalog`, conversion was made for the two new command mentioned above.